### PR TITLE
make the cursor work

### PR DESCRIPTION
**Pull request by acrisci**
Update to latest wlroots-rs, use the XCursorManager to make the cursor show up, and add pointer motion absolute event handler.

TODO:
* [ ] generalize cursor updates for relative/absolute motion into a single function

### CODE REVIEW

**Comment by Timidger** [comment]
Looks good, should be pretty easy to dry up that code. If there's issues you run into though feel free to point them out here

**Comment by acrisci** [comment]
Ok what do you think about that?

The idea here is to not pass handles to the helper functions. I think the other way could get a bit hairy because it increases the overhead of helper functions since they need to unwrap everything in the server each time you want to add a new one which could get nasty a few functions deep in the stack.

This way the only time you ever unwrap handles is at the wlroots entry point and everybody else works with the real things.

File: src/compositor/input/pointer.rs [hunk 1]
```diff
@@ -40,35 +48,52 @@ impl PointerHandler for Pointer {
...
+                    focus_under_pointer(seat, &mut **keyboards, None).expect("Could not focus view");
+                }
+            }).unwrap()
+        }).unwrap()
+    }
+}
+
+/// After the cursor has been warped, do all the things needed to change the cursor from the old
+/// position to the new position.
+fn update_cursor_position(cursor: &mut Cursor, seat: &mut Seat, views: &mut [View]) {
```
Comment (Timidger) [inline src/compositor/input/pointer.rs]: This doesn't actually update the cursor position though. It updates the views to be where the cursor is we should move the window (which in this proto-way-cooler phase it always is). If the idea is it should do other generic stuff then I think we shouldn't try to shoe horn those things into into this function and just focus on updating views.

This function name and doc comment should be changed to reflect that. (e.g. `move_focused_views_to_cursor` or something to that effect).

Comment (acrisci) [reply src/compositor/input/pointer.rs]: It will need to do other stuff too though like the resize.

Comment (Timidger) [reply src/compositor/input/pointer.rs]: Ok, but that isn't really updating the cursor position then is it? It should be `update_view_position`

Comment (acrisci) [reply src/compositor/input/pointer.rs]: We can call it that for now and then reevaluate the name later once you see the stuff that's going to go in there.

**Comment by Timidger** [comment]
Thanks!

**Pull request merged**